Packages/com.unity.inputsystem/Tests/TestFixture/ScopedDisposable.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using UnityEditor;
+
+namespace UnityEngine.InputSystem
+{
+    // Utility allowing access to object T as well as a dispose delegate to clean-up any resources associated with it.
+    // Useful with the 'using' keyword to provide scoped RAII-like cleanup of objects in tests without having a
+    // dedicated fixture handling the clean-up.
+    public sealed class ScopedDisposable<T> : IDisposable
```

Comment:
This doesn't seem to be used unless I'm missing something

Reply:
Correct, it was initially used but I managed to replace all its use with AssetDatabaseUtil.cs. I would like to keep it for now if OK since there are many use-cases for it in existing ill-formed tests that leak resources if any Assertion fails.

Reply:
It works really nice with C# 8 "using var", but doesn't seem to work well with Unity 2019 then... :/ it will be usable as soon as modern C# can be used in the tests since "using var scoped = Scoped.CreateObject<T>();" would auto clean-up regardless of where test execution was aborted and avoids writing tons of TearDown functions to handle it for each different scenario.